src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -1749,41 +1761,52 @@ private Matches() {}
     });
   }
 
-  public static final Match<Unit> UnitCreatesUnits = Match.of(unit -> {
-    final UnitAttachment ua = UnitAttachment.get(unit.getType());
-    if (ua == null) {
-      return false;
-    }
-    return (ua.getCreatesUnitsList() != null && ua.getCreatesUnitsList().size() > 0);
-  });
+  static Match<Unit> unitCreatesUnits() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getCreatesUnitsList() != null && ua.getCreatesUnitsList().size() > 0;
+    });
+  }
 
-  public static final Match<Unit> UnitCreatesResources = Match.of(unit -> {
-    final UnitAttachment ua = UnitAttachment.get(unit.getType());
-    if (ua == null) {
-      return false;
-    }
-    return (ua.getCreatesResourcesList() != null && ua.getCreatesResourcesList().size() > 0);
-  });
+  static Match<Unit> unitCreatesResources() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getCreatesResourcesList() != null && ua.getCreatesResourcesList().size() > 0;
+    });
+  }
 
-  public static final Match<UnitType> UnitTypeConsumesUnitsOnCreation = Match.of(unit -> {
-    final UnitAttachment ua = UnitAttachment.get(unit);
-    if (ua == null) {
-      return false;
-    }
-    return (ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0);
-  });
+  /**
+   * Returns a match indicating the specified unit type consumes at least one type of unit upon creation.
+   */
+  public static Match<UnitType> unitTypeConsumesUnitsOnCreation() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit);
+      if (ua == null) {
+        return false;
+      }
+      return ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0;
```

Comment:
The if check can be merged with this return statement: `return ua != null && ua.getConsumesUnits() != null && ua.getConsumesUnits().size() > 0;`